pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go

```diff
@@ -51,6 +53,23 @@ func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataS
 		return nil, err
 	}
 
+	if query.Headers == nil {
+		query.Headers = make(map[string]string)
+	}
+
+	if ds.JsonData != nil && ds.JsonData.Get("oauthPassThru").MustBool() {
+		// skip AddOAuthPassThruAuth for alerts
+		if query.User != nil {
+			token, err := social.GetCurrentOAuthToken(ctx, *query.User)
+			if err != nil {
+				tw.logger.Error("Error fetching oauth information for user", "error", err)
+			} else if token != nil {
+				delete(query.Headers, "Authorization")
+				query.Headers["Authorization"] = fmt.Sprintf("%s %s", token.Type(), token.AccessToken)
+			}
```

Comment:
```suggestion
			}
			delete(query.Headers, "Authorization")
			query.Headers["Authorization"] = fmt.Sprintf("%s %s", token.Type(), token.AccessToken)
```

I think in most cases, and `social.GetCurrentOauthToken` included, it's fair to assume that if `err` is nil, then the other value in the response is not.